lib/foreman_tasks/cleaner.rb

```diff
@@ -147,6 +148,15 @@ def delete_dynflow_plans_by_uuid(uuids)
       ForemanTasks.dynflow.world.persistence.delete_execution_plans({ 'uuid' => uuids }, batch_size, @backup_dir)
     end
 
+    def delete_orphaned_locks
+      orphaned_locks = ForemanTasks::Lock.left_outer_joins(:task).where(:'foreman_tasks_tasks.id' => nil)
+      with_noop(orphaned_locks, 'orphaned task locks') do |source, name|
+        with_batches(source, name) do |chunk|
+          ForemanTasks::Lock.where(id: chunk.map(&:id)).delete_all
```

Comment:
Good point about the `map` vs `pluck`, not sure what you mean by the select variant, but I prefer deleting the stuff in smaller chunks, rather than one big delete (see https://community.theforeman.org/t/testing-needed-improved-reports-expiration-rake-task/9856/1 for similar reasoning)

Reply:
To expand on the select idea:

By using `#pluck` you run a query to get the ids, load them into the ruby process, generate another query with the list of the ids and run it as a second query.

If you use `#select`, you run a single nested query (`delete from table where id in (select id from table where condition)`), without ever pulling the ids out into the ruby process.

Intuitively the second variant should perform better, but I don't have any data to back that up

Reply:
Go it: the problem is when I tried to do that, active record complained that it can't do delete with limit (maybe even postgres errored out). Using pluck with limit to keep the batch delete seems like a good compromise.

Reply:
Ok, fair enough